kernel/src/history_manager/timestamp_visitor.rs

```diff
@@ -0,0 +1,137 @@
+//! This module defines a visitor used to extract the in-commit timestamp from a commit file if it
+//! is present.
+use std::sync::{Arc, LazyLock};
+
+use crate::actions::COMMIT_INFO_NAME;
+use crate::expressions::column_name;
+use crate::schema::{ColumnNamesAndTypes, DataType, Schema, StructField, StructType};
+use crate::utils::require;
+use crate::{DeltaResult, Error, RowVisitor};
+
+/// This visitor extracts the in-commit timestamp (ICT) from a CommitInfo action if present. The
+/// [`EngineData`] being visited must have the schema defined in
+/// [`InCommitTimestampVisitor::schema`].
+///
+/// Only one row of the engine data is checked because CommitInfo. This is because in-commit
+/// timestamps requires that the CommitInfo containing the ICT be the first action in the log.
+#[allow(unused)]
+#[derive(Default)]
+pub(crate) struct InCommitTimestampVisitor {
+    pub(crate) in_commit_timestamp: Option<i64>,
```

Comment:
aside: maybe we should have a typedef for `type Timestamp = i64` (or name it something else that doesn't clash with our existing `DataType::Timestamp`)